components/QuestionBuilder/reducer.ts

```diff
@@ -0,0 +1,37 @@
+import { useReducer } from 'react';
+
+interface State {
+  textQuestions: string[];
+  selectedQuestion: number;
+}
+
+type Action =
+  | { type: 'add_text_question' }
+  | { type: 'select_question'; questionIndex: number };
```

Comment:
For right now, don't have too much logic for handling different question types, just text question. Would like to get the models in from #28